clickhouse/columns/column.h

```diff
@@ -42,11 +42,28 @@ class Column : public std::enable_shared_from_this<Column> {
     /// Appends content of given column to the end of current one.
     virtual void Append(ColumnRef column) = 0;
 
+    /// Template method to load column data from input stream. It'll call LoadPrefix and LoadBody.
+    /// Should be called only once from the client. Derived classes should not call it.
+    bool Load(InputStream* input, size_t rows);
+
+    /// Loads column prefix from input stream.
+    virtual bool LoadPrefix(InputStream* input, size_t rows);
+
     /// Loads column data from input stream.
-    virtual bool Load(InputStream* input, size_t rows) = 0;
+    virtual bool LoadBody(InputStream* input, size_t rows);
+
+    /// Saves column prefix to output stream. Column types with prefixes must implement it.
+    virtual void SavePrefix(OutputStream* output);
+
+    /// Saves column body to output stream.
+    virtual void SaveBody(OutputStream* output);
+
+    /// Saves column suffix to output stream. Column types with suffixes must implement it.
+    virtual void SaveSuffix(OutputStream* output);
 
-    /// Saves column data to output stream.
-    virtual void Save(OutputStream* output) = 0;
+    /// Template method to save to output stream. It'll call SavePrefix, SaveBody and SaveSuffix respectively
+    /// Should be called only once from the client. Derived classes should not call it.
+    virtual void Save(OutputStream* output);
```

Comment:
Why leaving `Save` a virtual method? Any of the derived columns may want to change serialization order?

Reply:
Will make it non virtual